components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultURLResolverService.java

```diff
@@ -40,7 +40,7 @@
 public class DefaultURLResolverService implements URLResolverService {
 
     @Override
-    public String resolveUrl(String url, boolean addProxyContextPath, boolean addWebContextRoot,
+    public String resolveUrl(String url, String tenantDomain, boolean addProxyContextPath, boolean addWebContextRoot,
```

Comment:
Shall we have a new API to accept tenant domain. 
Wondering if there is s real necessity of it as well.

In addition, we should have an API without legacy mode params as well. So new consumers don't need to worry about populating such.

Reply:
We have a new API for this at the IdentityUtil.class 
There's an another API available at IdentityUtil without legacy mode params.

There is a need for this because there are instances where tenantDomain is directly available as a attribute (for an instance, while tenant creation). So then, with this change, we don't need to meddle with the ThreadLocal or the CarbonContext to get the tenant domain.

Reply:
But the API should be clean.
So looks like this is addressed